x/auth/middleware/fee.go

```diff
@@ -4,78 +4,21 @@ import (
 	"context"
 	"fmt"
 
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/types/tx"
 	"github.com/cosmos/cosmos-sdk/x/auth/types"
 )
 
-var _ tx.Handler = mempoolFeeTxHandler{}
-
-type mempoolFeeTxHandler struct {
-	next tx.Handler
-}
-
-// MempoolFeeMiddleware will check if the transaction's fee is at least as large
```

Comment:
this should have a deprecation path. We should not add in such a large breaking change without allowing users to still use the old behaviour.

Reply:
If the user calls the `NewDefaultTxHandler`, then they only need to set `FeeMarket` to `authmiddleware.StaticFeeMarket{}`.
But if they were directly using `MempoolFeeMiddleware` or `TxPriorityMiddleware`, then they need to remove those and use `DeductFeeMiddleware` only.

Reply:
yes, I added the API breaking changes to the changelog now.

Reply:
amuary pointed out that middleware is already breaking and the priority is new to this release so its safe to break.